pkg/k8shandler/service.go

```diff
@@ -79,7 +79,7 @@ func (er *ElasticsearchRequest) CreateOrUpdateServices() error {
 	return nil
 }
 
-func createOrUpdateService(serviceName, namespace, clusterName, targetPortName string, port int32, selector, annotations map[string]string, publishNotReady bool, owner metav1.OwnerReference, labels map[string]string, client client.Client) error {
+func createOrUpdateService(serviceName, namespace, clusterName, targetPortName string, port int32, selector, annotations map[string]string, publishNotReady bool, labels map[string]string, cluster *loggingv1.Elasticsearch, client client.Client) error {
```

Comment:
Perhaps this PR merely reveals an existing problem but passing the `*loggingv1.Elasticsearch` struct to this function so that it can simply call `AddOwnerRefTo` seems excessive. I think this function should return the service to the caller and the caller should `AddOwnerRefTo` the result. This function has ten parameters which is already a code smell to me.

Reply:
You are right, this function is actually a method for `ElasticsearchRequest`. This simplifies the whole point that you want to see addressed while minimizing the refactoring to move this PR forward w/o much noise.

Reply:
Your observation highlights exactly why the request object was created and was supposed to encapsulate identifiers for things specific to a "cluster".  Ideally, the "cluster" object already knows about its name and namespace thus removing many of these arguments in favor of an "object oriented" approach.  It's the same reason `addOwnerRefTo` exists.